superchain/superchain.go

```diff
@@ -61,6 +71,251 @@ type AddressList struct {
 	ProxyAdmin                        Address `json:"ProxyAdmin"`
 }
 
+// ImplementationList represents the set of implementation contracts to be used together
+// for a network.
+type ImplementationList struct {
+	L1CrossDomainMessenger       VersionedContract `json:"L1CrossDomainMessenger"`
+	L1ERC721Bridge               VersionedContract `json:"L1ERC721Bridge"`
+	L1StandardBridge             VersionedContract `json:"L1StandardBridge"`
+	L2OutputOracle               VersionedContract `json:"L2OutputOracle"`
+	OptimismMintableERC20Factory VersionedContract `json:"OptimismMintableERC20Factory"`
+	OptimismPortal               VersionedContract `json:"OptimismPortal"`
+	SystemConfig                 VersionedContract `json:"SystemConfig"`
+}
+
+// ContractImplementations represent a set of contract implementations on a given network.
+// The key in the map represents the semantic version of the contract and the value is the
+// address that the contract is deployed to.
+type ContractImplementations struct {
+	L1CrossDomainMessenger       AddressSet `yaml:"l1_cross_domain_messenger"`
+	L1ERC721Bridge               AddressSet `yaml:"l1_erc721_bridge"`
+	L1StandardBridge             AddressSet `yaml:"l1_standard_bridge"`
+	L2OutputOracle               AddressSet `yaml:"l2_output_oracle"`
+	OptimismMintableERC20Factory AddressSet `yaml:"optimism_mintable_erc20_factory"`
+	OptimismPortal               AddressSet `yaml:"optimism_portal"`
+	SystemConfig                 AddressSet `yaml:"system_config"`
+}
+
+// AddressSet represents a set of addresses for a given
+// contract. They are keyed by the semantic version.
+type AddressSet map[string]Address
+
+// VersionedContract represents a contract that has a semantic version.
+type VersionedContract struct {
+	Version string  `json:"version"`
+	Address Address `json:"address"`
+}
+
+// Get will handle getting semantic versions from the set
+// in the case where the semver string is not prefixed with
+// a "v" as well as if it does have a "v" prefix.
+func (a AddressSet) Get(key string) Address {
+	if strings.HasPrefix(key, "v") {
+		if addr, ok := a[key]; ok {
+			return addr
+		}
+		if addr, ok := a[strings.TrimPrefix(key, "v")]; ok {
+			return addr
+		}
+	}
+	return a[key]
```

Comment:
[`strings.TrimPrefix`](https://pkg.go.dev/strings#TrimPrefix) "returns s without the provided leading prefix string. If s doesn't start with prefix, s is returned unchanged."

So I think we can do the following.
```suggestion
	if !strings.HasPrefix(key, "v") {
		key = "v" + key
	}
	if addr, ok := a[strings.TrimPrefix(key, "v")]; ok {
		return addr
	}
	return a[key]
```